### ISSUE TITLE
Whitespace messes with renaming

### ISSUE THREAD

**Issue by Zirak** [rollup/rollup#565]
Hello!

It seems like renaming and inlining has a bit of a whitespace issue. Minimal example:

```js
// entry.js
import * as foo from './foo';
foo .yar();

// foo.js
export function yar() {
    console.log('yar?');
}

// build: rollup entry.js
```

And the result is
```js
function yar() {
    console.log('yar?');
}

yarr(); // notice the extra r
```

Or for a more noticeable difference:

```js
// entry.js
// ...
foo
    .yar();

// output
// ...
yar .yar();
```

Play with the whitespace count for more amusing results.

Digging a bit got me to [`Module.js`](https://github.com/rollup/rollup/blob/b50167646efe19297b40de49f19caebe2eb77018/src/Module.js#L516). Debugging shows that  `magicString.original.slice(start, end)` doesn't capture the entire statement, the whitespace is ignored in calculating the `end` offset.

Since the `start` and `end` values come from the AST, I believe this is an acorn issue and may have to be fixed upstream.

Thoughts? Maybe this could somehow be patched in rollup without waiting for upstream?

**Comment by Zirak** [comment]
On closer inspection this isn't an acorn issue. Playing with it provided the expected results, and putting a breakpoint [here](https://github.com/rollup/rollup/blob/b50167646efe19297b40de49f19caebe2eb77018/src/Module.js#L297) just after we get the ast and inspecting it shows correct offsets. It gets jumbled in the way somewhere. I'll see if I can track down where.

**Comment by eventualbuddha** [comment]
@Zirak that's what I found, too. I added your example as a test in rollup's test suite and it behaves as you describe. I'll try to track it down too, but let me know if you find it first.

**Comment by Zirak** [comment]
![I think I've got it](http://cdn.staticneo.com/w/layton/thumb/1/10/LukeCorrect.png/180px-LukeCorrect.png)

Found it. The range of references are declared [in `Reference.js`](https://github.com/rollup/rollup/blob/b50167646efe19297b40de49f19caebe2eb77018/src/Reference.js#L19-L20), and are then modified [in `SyntheticNamespaceDeclaration`](https://github.com/rollup/rollup/blob/51a87a85c9c8f9b11b65cf08f713367fd195193c/src/Declaration.js#L198-L200), where we find a quite appropriate `TODO`. Only the names and the periods separating names are taken into account, not the whitespace.

**Comment by eventualbuddha** [comment]
Thanks for the help, @Zirak. This should be fixed in #566.

**Comment by Zirak** [comment]
Neat, thanks! I was starting to play with `Reference#parts`, making it an array of nodes instead of strings. You seem to have saved me some time.

**Comment by eventualbuddha** [comment]
[Fixed](http://rollupjs.org/#%7B%22options%22%3A%7B%22format%22%3A%22cjs%22%2C%22moduleName%22%3A%22myBundle%22%2C%22globals%22%3A%7B%7D%7D%2C%22modules%22%3A%5B%7B%22name%22%3A%22main.js%22%2C%22code%22%3A%22import%20*%20as%20foo%20from%20'.%2Ffoo.js'%3B%5Cnfoo%20.yar()%3B%22%7D%2C%7B%22name%22%3A%22foo.js%22%2C%22code%22%3A%22export%20function%20yar()%20%7B%5Cn%20%20%20%20console.log('yar%3F')%3B%5Cn%7D%22%7D%5D%7D) in v0.25.5.

**Comment by Zirak** [comment]
I just realised this causes [another issue:]( http://rollupjs.org/#%7B%22options%22%3A%7B%22format%22%3A%22cjs%22%2C%22moduleName%22%3A%22myBundle%22%2C%22globals%22%3A%7B%7D%7D%2C%22modules%22%3A%5B%7B%22name%22%3A%22main.js%22%2C%22code%22%3A%22import%20*%20as%20foo%20from%20'.%2Ffoo.js'%3B%5Cnfoo.yar.har()%3B%22%7D%2C%7B%22name%22%3A%22foo.js%22%2C%22code%22%3A%22export%20function%20yar()%20%7B%5Cn%20%20%20%20return%20%7B%5Cn%20%20%20%20%20%20har()%20%7B%5Cn%20%20%20%20%20%20%20%20console.log('yar%3F')%3B%5Cn%20%20%20%20%20%20%7D%5Cn%20%20%20%20%7D%5Cn%7D%22%7D%5D%7D)

```js
// entry.js
import * as foo from './foo.js';
foo.yar.har();

// foo.js
export function yar() {
    return {
      har() {
        console.log('yar?');
      }
    }
}

// result:
function yar() {
    return {
      har() {
        console.log('yar?');
      }
    }
}

yar(); // no har
```

**Comment by eventualbuddha** [comment]
Fully fixed by v0.25.6.
